components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -253,21 +253,57 @@ fn sqrt(arg: &Option<Real>) -> Result<Option<Real>> {
     }))
 }
 
+#[inline]
+#[rpn_fn]
+fn sin(arg: &Option<Real>) -> Result<Option<Real>> {
+    Ok(arg.map(|arg| Real::from(arg.sin())))
```

Comment:
I would prefer to use `Real::new()` so that invalid values are mapped to `None` instead of panic.

Reply:
Okay, that sounds better